src/lib/tabwidget/combotabbar.cpp

```diff
@@ -534,8 +534,9 @@ void ComboTabBar::setUpLayout()
     }
 
     // We need to setup heights even before m_mainTabBar->height() has correct value
-    // So lets just set minimum 5px height
-    height = qMax(5, height);
+    // So lets just set minimum 32px height so that if just pinned tabs were present
+    // in the last session, then height becomes very small
+    height = qMax(32, height);
```

Comment:
I'm not really convinced by this. Although it does "fix" the issue in most cases, it is still just hardcoded value and will break in case where the real tabbar height is much bigger (so 32 is still too small).